lib/DxilPIXPasses/DxilDbgValueToDbgDeclare.cpp

```diff
@@ -399,12 +419,212 @@ static OffsetInBits GetAlignedOffsetFromDIExpression(
   return Exp->getBitPieceOffset();
 }
 
+llvm::DISubprogram* GetFunctionDebugInfo(llvm::Module& M, llvm::Function* fn) {
+  auto FnMap = makeSubprogramMap(M);
+  return FnMap[fn];
+}
+
+GlobalVariableToLocalMirrorMap
+GenerateGlobalToLocalMirrorMap(llvm::Module &M,
+                                    llvm::DIGlobalVariable *DIGV) {
+  hlsl::DxilModule &DM = M.GetOrCreateDxilModule();
+  auto entryPoints = DM.GetExportedFunctions();
+
+  std::string LocalMirrorOfGlobalName =
+      std::string("global.") + std::string(DIGV->getName());
+
+  GlobalVariableToLocalMirrorMap ret;
+  DenseMap<const Function *, DISubprogram *> FnMap;
+
+  for (llvm::Function const *fn : entryPoints) {
+    auto & LocalMirror = ret[fn];
+    auto &blocks = fn->getBasicBlockList();
+    for (auto &block : blocks) {
+      bool breakOut = false;
+      for (auto &instruction : block) {
+        if (auto const *DbgValue = llvm::dyn_cast<llvm::DbgValueInst>(&instruction)) {
```

Comment:
Do we just need to check `DbgValueInst`? What about `DbgDeclareInst`?

Reply:
If we alos need to check `DbgDeclareInst`, we could just dyn_cast to `DbgInfoIntrinsic`.

Reply:
Good point. I added a check for DbgDeclare too. (I need the variable, so DbgInfoIntrinsic isn't for enough down the class hierarchy.)